gneiss/_summary.py

```diff
@@ -57,3 +58,72 @@ def __init__(self, stat_results,
         # calculate the overall coefficient of determination (i.e. R2)
         sst = sse + ssr
         self.r2 = 1 - sse / sst
+
+    def _check_projection(self, project):
+        """
+        Parameters
+        ----------
+        project : bool
+           Specifies if a projection into the Aitchison simplex can be
+           performed.
+
+        Raises
+        ------
+        ValueError:
+            Cannot perform projection into Aitchison simplex if `basis`
+            is not specified.
+        ValueError:
+            Cannot perform projection into Aitchison simplex
+            if `feature_names` is not specified.
+        """
+        if self.basis is None and project:
+            raise ValueError("Cannot perform projection into Aitchison simplex"
+                             "if `basis` is not specified.")
+
+        if self.feature_names is None and project:
+            raise ValueError("Cannot perform projection into Aitchison simplex"
+                             "if `feature_names` is not specified.")
+
+    def coefficients(self, project=False):
+        """ Returns coefficients from fit.
+
+        Parameters
+        ----------
+        project : bool, optional
+            Specifies if coefficients should be projected back into
+            the Aitchison simplex.  If false, the coefficients will be
+            represented as balances  (default: False).
+
+        Returns
+        -------
+        pd.DataFrame
+            A table of values where columns are coefficients, and the index
+            is either balances or proportions, depending on the value of
+            `project`.
+
+        Raises
+        ------
+        ValueError:
+            Cannot perform projection into Aitchison simplex if `basis`
+            is not specified.
+        ValueError:
+            Cannot perform projection into Aitchison simplex
+            if `feature_names` is not specified.
+        """
+        self._check_projection(project)
+        coef = pd.DataFrame()
+
+        for r in self.results:
+            c = r.params
+            c.name = r.model.endog_names
+            coef = coef.append(c)
+
+        if project:
+            # `check=True` due to type issue resolved here
+            # https://github.com/biocore/scikit-bio/pull/1396
+            c = ilr_inv(coef.values.T, basis=self.basis, check=False).T
```

Comment:
Ok. Then this should be ok

On Jul 31, 2016 6:38 AM, "Antonio Gonzalez" <notifications@github.com>
wrote:

> In gneiss/_summary.py
> <https://github.com/biocore/gneiss/pull/28#discussion_r72904299>:
>
> > +        ValueError:
> > +            Cannot perform projection into Aitchison simplex
> > +            if `feature_names` is not specified.
> > +        """
> > +        self._check_projection(project)
> > +        coef = pd.DataFrame()
> > +
> > +        for r in self.results:
> > +            c = r.params
> > +            c.name = r.model.endog_names
> > +            coef = coef.append(c)
> > +
> > +        if project:
> > +            # `check=True` due to type issue resolved here
> > +            # https://github.com/biocore/scikit-bio/pull/1396
> > +            c = ilr_inv(coef.values.T, basis=self.basis, check=False).T
>
> That fix has been merged in skbio and you could use the dev version of
> skbio (in travis, setup.py, etc) until there is release, will this work?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/biocore/gneiss/pull/28/files/f3b7d9472d3eb4ebebe004b42141f66d20f9461a#r72904299>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AD_a3Tyk2nu8mJS-28A2xkXGhFb67ZI_ks5qbKU6gaJpZM4JXppU>
> .
>